### DISCUSSION TITLE
My Trick of Deciding if a URL Should Open a New Browser Tab | /en/2018/09/target-blank/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#1208] [Comments]
https://yihui.org/en/2018/09/target-blank/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks! This method has two limitations: 1. It only works with the Markdown renderer Blackfriday (e.g. on Hugo websites); 2. It adds target="_blank" to all links in the body of the post. The JS solution I provided is a general-purpose solution that works on all links on any HTML pages.

> Originally posted on 2018-09-22 02:27:37

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Clever idea! I did this in JavaScript for the rmarkdown website: <https://github.com/rstudio/rmarkdown/blob/dc278e4753b27991e9878f1d60b0c32f9a2ffa0d/js/external-links.js#L9>

> Originally posted on 2019-08-08 14:06:03

**Reply by IdoBar** [reply]
Thanks for this suggestion, but I couldn't understand where to put this code in an Rmarkdown file for it to change all my links.

[a comment by an automated account is not shown]
